src/elysium/wallet.h

```diff
@@ -46,26 +56,45 @@ class Wallet
 
     void ClearAllChainState();
 
-    SigmaSpend CreateSigmaSpend(PropertyId property, SigmaDenomination denomination, bool fPadding);
+    SigmaSpend CreateSigmaSpendV0(PropertyId property, SigmaDenomination denomination, bool fPadding);
+    SigmaSpend CreateSigmaSpendV1(PropertyId property, SigmaDenomination denomination, bool fPadding);
+
     void DeleteUnconfirmedSigmaMint(SigmaMintId const &id);
 
 public:
     template<class OutputIt>
-    void ListSigmaMints(OutputIt it)
+    void ListSigmaMintsV0(OutputIt it)
+    {
+        mintWalletV0.ListMints(it);
+    }
+
+    template<class OutputIt>
+    void ListSigmaMintsV1(OutputIt it)
     {
-        mintWallet.ListMints(it);
+        mintWalletV1.ListMints(it);
     }
 
     SigmaMint GetSigmaMint(const SigmaMintId& id);
+    CoinSigner GetSigmaSigner(const SigmaMintId &id);
     SigmaPrivateKey GetKey(const SigmaMint &mint);
+
     bool HasSigmaMint(const SigmaMintId& id);
     bool HasSigmaMint(const secp_primitives::Scalar &serial);
 
 protected:
     boost::optional<SigmaMint> GetSpendableSigmaMint(
-        PropertyId property, SigmaDenomination denomination);
+        PropertyId property, SigmaDenomination denomination, CoinVersion version);
     void SetSigmaMintChainState(const SigmaMintId &id, const SigmaMintChainState &state);
 
+    SigmaWallet& GetMintWallet(CoinVersion version);
+    SigmaWallet& GetMintWallet(SigmaMintId const &id);
+    SigmaWallet& GetMintWallet(secp_primitives::Scalar &serial);
+
+    bool HasSigmaMint(const SigmaMintId& id, CoinVersion &version);
```

Comment:
This one is quite hard to use. My idea is removing the last argument and make it returns `boost::optional<CoinVersion>` instead.

Reply:
Good idea